docs/readmes/orc8r/dev_sub_digests.md

```diff
@@ -0,0 +1,88 @@
+
+## Motivation
+
+Previously, orc8r follows a vanilla data transmission pattern that sends all subscriber data down to all gateways at every configurable interval.
+
+This introduces various issues at scale, a major one being the significant amount of network pressure this pattern necessitates. For example, to achieve our [GA scalability targets](https://docs.google.com/document/d/1P306DmuC1CFi7bqz4_VVi7v4Ewr1y-dD47C-s5XzdJ8/edit), we need to support **~20TB of network load per month per network**, which is definitively unsustainable.
+
+Therefore, starting from v1.6, as part of our [subscriber scaling project](../proposals/p010_subscriber_scaling.md), orc8r implements the subscriber digests pattern to minimize network pressure and optimize the current mode of subscriber data transmission. This pattern utilizes the assumption of reads >> writes, and focuses on two major improvements
+1. Instead of syncing at every interval, only sync when AGW is out-of-sync
+2. Instead of syncing all data, sync the cloud-AGW data changeset when reasonable
+
+## Overview
+
+The architectural additions implemented for the subscriber digests pattern are highlighted below.
+
+![Subscriber Digests Architecture](assets/orc8r/subscriber_digests_architecture.png)
+
+### Digest Generation
+
+This pattern relies on taking deterministically encoded data snapshots, a.k.a. "digests", that are used to represent the local data version. To this end, orc8r library code `mproto` is created using the golang third-party library `proto` to conduct consistently deterministic data serializations.
+
+The subscriber digests pattern specifically generates and utilizes two types of digests
+
+1. **Flat digests**: a digest of the entire set of subscriber data of a network
+2. **Per-subscriber digests**: a digest of a single subscriber data object; for a network, the list of per-subscriber digests is tracked and distributed en masse
+
+NOTE: Currently, a decoupling process between subscriber data objects and their gateway-specific APN resource configurations is conducted to make sure the generated digests is representative and also network-general. See [additional notes](./dev_sub_digests.md#apn-resource-handling) for more details.
+
+### Subscriber Digests Cache
+`subscriberdb_cache` is a single-pod service in sole charge of managing network digests. The service consists mainly of worker code that constantly generates new flat digests and per-subscriber digests at a configurable interval for each network.
+
+The generated digests are written to a cloud SQL store to which `subscriberdb` service has read-only access. In this way, `subscriberdb` can directly read the most recently cached digests from the SQL store, instead of making extraneous gRPC calls over the network and having multiple servicer workers compete for the `subscriberdb_cache` endpoints.
```

Comment:
```suggestion
The generated digests are written to a cloud SQL store to which `subscriberdb` has read-only access. In this way, `subscriberdb` can directly read the most up-to-date digests from the SQL store, instead of making extraneous gRPC calls over the network and contending for the limited resources of the `subscriberdb_cache` singleton.
```